odl/discr/partition.py

```diff
@@ -546,6 +555,139 @@ def squeeze(self):
         newset = self.set[self.grid.inondeg]
         return RectPartition(newset, self.grid.squeeze())
 
+    def index(self, value, floating=False):
+        """Return the index of a value in the domain.
+
+        Parameters
+        ----------
+        value : ``self.set`` element
+            The point to find the index of.
+        floating : bool
+            True if the index should also give the position inside the voxel.
+            This is given by returning the integer valued index of the voxel
+            plus the partial distance from the left to the right boundary.
+
+        Returns
+        -------
+        index : int/float or tuple of int/float
```

Comment:
That's what I meant, i.e. the suggestion applied to the second part only. Another possibility would be
```
index : int, float or tuple of these
```
but it's not quite as good.